requesttojson/requesttojson_test.go

```diff
@@ -0,0 +1,251 @@
+}
+
+func TestRequestToJSON_WithHeaders(t *testing.T) {
+	// Create a POST request with headers
+	req, err := http.NewRequest("POST", "http://example.com/path", nil)
+	require.NoError(t, err)
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("X-Custom-Header", "CustomValue")
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check the headers in struct
+	require.Equal(t, map[string][]string{
+		"Content-Type":    {"application/json"},
+		"X-Custom-Header": {"CustomValue"},
+	}, reqJSON.Headers)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"Content-Type":["application/json"]`)
+	require.Contains(t, string(jsonData), `"X-Custom-Header":["CustomValue"]`)
+}
+
+func TestRequestToJSON_WithBody(t *testing.T) {
+	// Create a POST request with a body
+	body := `{"key": "value"}`
+	req, err := http.NewRequest("POST", "http://example.com/path", strings.NewReader(body))
+	require.NoError(t, err)
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check the body
+	require.Equal(t, body, reqJSON.Body)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"body":"{\"key\": \"value\"}"`)
+}
+
+func TestRequestToJSON_WithQueryParams(t *testing.T) {
+	// Create a GET request with query parameters
+	req, err := http.NewRequest("GET", "http://example.com/search?q=golang&sort=asc", nil)
+	require.NoError(t, err)
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check the query parameters
+	expectedQuery := map[string][]string{
+		"q":    {"golang"},
+		"sort": {"asc"},
+	}
+	require.Equal(t, expectedQuery, reqJSON.Query)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"query_parameters":{"q":["golang"],"sort":["asc"]}`)
+}
+
+func TestRequestToJSON_EmptyBodyReplaced(t *testing.T) {
+	// Create a POST request with an empty body
+	req, err := http.NewRequest("POST", "http://example.com/path", nil)
+	require.NoError(t, err)
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "{}")
+	require.NoError(t, err)
+
+	// Check the body is empty
+	require.Equal(t, reqJSON.Body, "{}")
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"body":"{}"`)
+}
+
+func TestRequestToJSON_EmptyBody(t *testing.T) {
+	// Create a POST request with an empty body
+	req, err := http.NewRequest("POST", "http://example.com/path", nil)
+	require.NoError(t, err)
+
+	// Call the function
+	jsonData, reqJSON, err := RequestToJSON(req, "")
+	require.NoError(t, err)
+
+	// Check the body is empty
+	require.Empty(t, reqJSON.Body)
+
+	// Check the JSON output
+	require.Contains(t, string(jsonData), `"body":""`)
+}
+
+func TestRequestToJSON_VariousHTTPMethods(t *testing.T) {
+	methods := []string{"PUT", "DELETE", "PATCH", "OPTIONS", "HEAD"}
```

Comment:
We can use similar set of constants at other places as well.
```suggestion
	methods := []string{http.MethodPut, http.MethodDelete, http.MethodPatch, http.MethodOptions, http.MethodHead}
```